### PR TITLE
fix: Track types and Improve Perf & Convert Node methods into async.

### PR DESCRIPTION
**Pull request by Joniii11**
So i fixed some more stuff and also optimized a bit.

## `Node.ts`:
- `<Node>.message` Made async, implemented a switch statement. Fixed types for type safety
- `<Node>.reconnect` Made async
- Changed some parts of code to make it look a bit nicer to look at
- `<Node>.open` implemented try/catch to catch possible rejections from any of the awaitable functions
- `<Node>.close` Made async

## `Track.ts`:
- Changed the types
- Improved the destructurlizing of the objects to maybe prevent that something isn't undefined.

I will add more changes on the way

### CODE REVIEW

File: src/Node/Node.ts [hunk 1]
```diff
@@ -250,48 +293,71 @@ export class Node {
...
+                        await this.rest.patch(`/v4/sessions/${this.sessionId}`, { resumingKey: this.resumeKey, timeout: this.resumeTimeout });
+                        this.poru.emit("debug", this.name, `[Lavalink Rest]  Resuming configured on Lavalink`);
+                    };
+
+                    break;
+                };
+
+                // If the packet has stats about the node in it update them on the Node's class
+                case "stats": {
+                    this.stats = packet
```
Comment (UnschooledGamer) [inline src/Node/Node.ts]: `Node#stats` will have `packet.op` also if not deleted

Comment (Joniii11) [reply src/Node/Node.ts]: To be honest it doesn't change anything but ig sure

Comment (UnschooledGamer) [reply src/Node/Node.ts]: It doesn't, While the user will access `Node#stats` it would display `op` also.

Comment (UnschooledGamer) [reply src/Node/Node.ts]: @Joniii11 ?

Comment (Joniii11) [reply src/Node/Node.ts]: I was sleeping

Comment (UnschooledGamer) [reply src/Node/Node.ts]: Oh

Comment (Joniii11) [reply src/Node/Node.ts]: Is fixed now :D

**Pull request merged**